flux/stdlib/influxdata/influxdb/to.go

```diff
@@ -147,10 +186,57 @@ func NewToTransformation(ctx context.Context, d execute.Dataset, cache execute.T
 	if err != nil {
 		return nil, err
 	}
+	var fn *execute.RowMapFn
+	if spec.FieldFn.Fn != nil {
+		fn = execute.NewRowMapFn(spec.FieldFn.Fn, compiler.ToScope(spec.FieldFn.Scope))
+	}
+	var isTagColumn func(column flux.ColMeta) bool
+	if toSpec.Spec.TagColumns == nil {
+		// If no tag columns are specified, by default we exclude
+		// _field, _value and _measurement from being tag columns.
+		excludeColumns := map[string]bool{
+			execute.DefaultValueColLabel: true,
+			DefaultFieldColLabel:         true,
+			DefaultMeasurementColLabel:   true,
+		}
+
+		// Also exclude the overridden measurement column
+		excludeColumns[toSpec.Spec.MeasurementColumn] = true
+
+		// If a field function is specified then we exclude any column that
+		// is referenced in the function expression from being a tag column.
+		if toSpec.Spec.FieldFn.Fn != nil {
+			recordParam := toSpec.Spec.FieldFn.Fn.Parameters.List[0].Key.Name
+			exprNode := toSpec.Spec.FieldFn.Fn
+			colVisitor := newFieldFunctionVisitor(recordParam)
+
+			// Walk the field function expression and record which columns
+			// are referenced. None of these columns will be used as tag columns.
+			semantic.Walk(colVisitor, exprNode)
+			for k, v := range colVisitor.captured {
```

Comment:
Note that 2.x delays this processing to `Process` so it can evaluate the captured columns against the actual table columns. That seems unnecessary to me, much nicer to do it up front.